File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double max = 0;
        for(int i=0;i<k;i++){
            max+=nums[i];
        }
        double currSum = max;
        for(int i=k;i<nums.length;i++){
            currSum+=nums[i]-nums[i-k];
            if(max<currSum){
                max = currSum;
            }
        }
        return (double) max/k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        curr = sum(nums[:k])
        top = curr

        for i in range(k, len(nums)):
            curr = curr + nums[i] - nums[i -k]
            if curr > top:
                top  = curr
        return top/k